CourierAppBackend/src/main/java/com/fdmgroup/courierapp/controller/AccountController.java

```diff
@@ -43,16 +44,24 @@ public class AccountController {
 
     @PostMapping("/register")
     public ResponseEntity<ResponseRegister> registerSender(@RequestBody RequestRegister requestRegister) {
-        //check duplicate email
-        if (customerService.isDuplicateEmail(requestRegister.getEmail())){
-            ResponseRegister response = new ResponseRegister("Failed", "Duplicate Email");
+        //check duplicate email and username
+        Boolean isDuplicateEmail = customerService.isDuplicateEmail(requestRegister.getEmail());
+        Boolean isDuplicateUsername = accountService.isDuplicateUsername(requestRegister.getUsername());
+        if (isDuplicateUsername && isDuplicateEmail) {
```

Comment:
For now ok, if we have more exceptions next time, append to list and return 1 shot better. (aka reduce duplicated code)